R/orderlyweb.R

```diff
@@ -199,6 +197,35 @@ R6_orderlyweb <- R6::R6Class(
       value
     },
 
+    bundle_pack = function(name, parameters = NULL, instance = NULL,
+                           progress = TRUE) {
+      parameters <- report_run_parameters(parameters)
+      if (!is.null(instance)) {
+        query <- list(instance = instance)
+      } else {
+        query <- NULL
+      }
+      download <- orderlyweb_download(tempfile(), progress, "zip")
+      res <- self$api_client$POST(sprintf("/bundle/pack/%s/", name),
+                                  body = parameters, download = download,
+                                  encode = "json", query = query)
+      fix_progress_print(progress)
+
+      filename <- paste0(sub("/.*", "", zip::zip_list(res)$filename[[1]]),
+                         ".zip")
+
+      dest <- file.path(tempdir(), filename)
+      fs::file_move(res, dest)
+
+      dest
+    },
+
+    bundle_import = function(path, progress = TRUE) {
+      self$api_client$POST("/bundle/import/",
+                           body = httr::upload_file(path),
+                           if (progress) httr::progress("up"))
```

Comment:
Should `fix_progress_print` be used here too?

Reply:
The httr progress bar fails to print a trailing newline after download of an object of unknown length (probably this is something we should fix in the headers coming from orderly.server or OrderlyWeb?). But here, httr knows the length so we get a progress bar that cleans up properly.

Without the fix where it is needed, the user will see their REPL prompt dumped half way across the screen...